### PR TITLE
Add OfType to static semantics

### PR DESCRIPTION
**Pull request by qurben**

### CODE REVIEW

File: stratego.gradual/trans/static-semantics.stx [hunk 1]
```diff
@@ -272,6 +280,10 @@ rules // rules
     
   typeOfBuildTerm(s, CastTerm(e, t1, t2)) = (CastTerm(e, t1, t2), t). // Silence CastTerm
   
+  typeOfBuildTerm(s, OfType(e, t)) = (CastTerm(e', t', t), t) :-
```
Comment (fletna) [inline stratego.gradual/trans/static-semantics.stx:283]: I think you should do this via cast and not directly create CastTerm

File: stratego.gradual/trans/static-semantics.stx [hunk 2]
```diff
@@ -216,6 +217,9 @@ rules // rules
   assertTypeOfMatchTerm(s, Tuple(trms)) = TUP(ts) :-
   	assertTypeOfMatchTerms(s, trms) == ts.
   	
+  assertTypeOfMatchTerm(s, OfType(e, t)) = t' :-
```
Comment (fletna) [inline stratego.gradual/trans/static-semantics.stx:220]: why t prime?

**Review by fletna — COMMENTED** [review]
Are we allowing type hints on matched terms in rules?

**Comment by qurben** [comment]
Closing in favour of #8

**Pull request closed without merging**